File: src/Cryptofolio.App/ClientApp/src/app/models/currency.ts
export class Currency {
  id: string;
  name: string;
  code: string;
  symbol: string;
  precision: number;
  valueFormat: string;

  constructor(properties?: any) {
    Object.assign(this, properties);
    this.valueFormat = properties["value_format"];
  }
}

export class CurrencyTicker {
  currency: Currency;
  vsCurrency: Currency;
  timestamp: Date;
  value: number;

  constructor(properties?: any) {
    Object.assign(this, properties);
    this.currency = new Currency(properties.currency);
    this.vsCurrency = new Currency(properties["vs_currency"]);
    this.timestamp = new Date(properties.timestamp);
  }
}

File: src/Cryptofolio.Infrastructure/Entities/Currency.cs
using System.Text.Json.Serialization;

namespace Cryptofolio.Infrastructure.Entities
{
    /// <summary>
    /// Models a fiat currency.
    /// </summary>
    public class Currency
    {
        /// <summary>
        /// The id.
        /// </summary>
        public string Id { get; set; }

        /// <summary>
        /// The name.
        /// </summary>
        public string Name { get; set; }

        /// <summary>
        /// The code.
        /// </summary>
        public string Code { get; set; }

        /// <summary>
        /// The symbol.
        /// </summary>
        public string Symbol { get; set; }

        /// <summary>
        /// The roungind precision.
        /// </summary>
        public int Precision { get; set; } = 2;

        /// <summary>
        /// The value format.
        /// </summary>
        [JsonPropertyName("value_format")]
        public string ValueFormat { get; set; }
    }
}
